synapse/federation/transaction_queue.py

```diff
@@ -69,13 +70,20 @@ def __init__(self, hs, transport_layer):
         # destination -> list of tuple(edu, deferred)
         self.pending_edus_by_dest = edus = {}
 
+        self.pending_presence_by_dest = presence = {}
+        self.pending_edus_keyed_by_dest = edus_keyed = {}
```

Comment:
Because presence EDU's are a combination of multiple users presence, so we need to manually create the EDU at send time.